src/EditorFeatures/Core/Implementation/NavigationBar/NavigationBarController.cs

```diff
@@ -153,81 +134,26 @@ public void Disconnect()
-            if (args.NewSolution.Workspace != _workspace)
-            {
-                return;
-            }
-
-            // If the displayed project is being renamed, retrigger the update
-            if (args.Kind == WorkspaceChangeKind.ProjectChanged && args.ProjectId != null)
-            {
-                var oldProject = args.OldSolution.GetRequiredProject(args.ProjectId);
-                var newProject = args.NewSolution.GetRequiredProject(args.ProjectId);
-
-                if (oldProject.Name != newProject.Name)
-                {
-                    var currentContextDocumentId = _workspace.GetDocumentIdInCurrentContext(_subjectBuffer.AsTextContainer());
-
-                    if (currentContextDocumentId != null && currentContextDocumentId.ProjectId == args.ProjectId)
-                    {
-                        StartModelUpdateAndSelectedItemUpdateTasks(modelUpdateDelay: 0);
-                    }
-                }
-            }
-
-            if (args.Kind == WorkspaceChangeKind.DocumentChanged &&
-                args.OldSolution == args.NewSolution)
-            {
-                var currentContextDocumentId = _workspace.GetDocumentIdInCurrentContext(_subjectBuffer.AsTextContainer());
-                if (currentContextDocumentId != null && currentContextDocumentId == args.DocumentId)
-                {
-                    // The context has changed, so update everything.
-                    StartModelUpdateAndSelectedItemUpdateTasks(modelUpdateDelay: 0);
-                }
-            }
-        }
-
-        private void OnDocumentActiveContextChanged(object? sender, DocumentActiveContextChangedEventArgs args)
-        {
-            AssertIsForeground();
-            if (args.Solution.Workspace != _workspace)
-                return;
-
-            var currentContextDocumentId = _workspace.GetDocumentIdInCurrentContext(_subjectBuffer.AsTextContainer());
-            if (args.NewActiveContextDocumentId == currentContextDocumentId ||
-                args.OldActiveContextDocumentId == currentContextDocumentId)
-            {
-                // if the active context changed, recompute the types/member as they may be changed as well.
-                StartModelUpdateAndSelectedItemUpdateTasks(modelUpdateDelay: 0);
-            }
-        }
-
-        private void OnSubjectBufferPostChanged(object? sender, EventArgs e)
-        {
-            AssertIsForeground();
-            StartModelUpdateAndSelectedItemUpdateTasks(modelUpdateDelay: TaggerConstants.MediumDelay);
-        }
-
         private void OnCaretMoved(object? sender, EventArgs e)
         {
             AssertIsForeground();
-            StartSelectedItemUpdateTask(delay: TaggerConstants.NearImmediateDelay);
+            StartSelectedItemUpdateTask();
         }
 
         private void OnViewFocused(object? sender, EventArgs e)
         {
             AssertIsForeground();
-            StartSelectedItemUpdateTask(delay: TaggerConstants.ShortDelay);
```

Comment:
the need for this fine-grained delays was not really necessary.  We need to do the work anyways, and the user isn't going to notice us cmoputing things 50 vs 200 ms later for things like navbars.